lib/server/services/run/updateRun.js

```diff
@@ -114,8 +114,8 @@ exports.updateRun = async (identifier, payload, transaction) => {
             const { runQualityChangeReason } = metadata;
 
             const { lhcPeriodId } = runModel;
-            const lhcPeriod = await getOrCreateLhcPeriod(lhcPeriodId);
-            const lhcPeriodName = lhcPeriod.name;
+            const lhcPeriod = await getLhcPeriodOrFail(lhcPeriodId);
```

Comment:
As this method can throw, it means that your UpdateRun will fail as well due to this.
As we do not want that, you should either:
- use a `try/catch` in here and update the period name accordingly
- or, you should use the `getLhcPeriod` log which will return null instead of throwing error. 

I think you should go with second option as we are interested in continuing the log creation even if period name cannot be found.